snippets/flatMap.md

```diff
@@ -0,0 +1,19 @@
+---
+title: flatMap
+tags: array,function,intermediate
+---
+
+The flatMap() method first maps each element using a mapping function, then flattens the result into a new array. It is identical to a map() followed by a flat() of depth 1, but flatMap() is often quite useful, as merging both into one method is slightly more efficient.
```

Comment:
Please use `Array.prorotype.map()` instead of `map()` in the description. 👍 We're trying to be consistent with naming built-in functions.